app/tui/side_bar.py

```diff
@@ -1,74 +1,51 @@
 from __future__ import unicode_literals
+from typing import Any, List
 
-from typing import Any, Callable, Optional
-
-from prompt_toolkit.application import get_app
 from prompt_toolkit.buffer import Buffer
 from prompt_toolkit.key_binding import KeyBindings
 from prompt_toolkit.layout import DynamicContainer, HSplit, ScrollbarMargin, Window
 from prompt_toolkit.layout.controls import BufferControl, FormattedTextControl
 from prompt_toolkit.layout.dimension import D, Dimension
-from prompt_toolkit.mouse_events import MouseEvent
 from prompt_toolkit.widgets import Frame
 
-from app.context import ProcMuxContext
-from app.log import logger
-from app.tui.focus import FocusManager
-from app.tui.keybindings import register_app_wide_configured_keybindings, register_configured_keybinding
-from app.tui_state import FocusWidget
+from app.tui.controller import ProcMuxController
+from app.tui.keybindings import register_app_wide_configured_keybindings, register_configured_keybinding_no_event
+from app.tui.state import FocusWidget
 
 
 class SideBar:
+    _buffer_control: BufferControl
+    _container: Frame
+    _controller: ProcMuxController
+    _filter_buffer: Buffer
+    _fixed_width: int
+    _list_control: FormattedTextControl
```

Comment:
im curious, by defining the variable here, doesnt it become attached the class instead of the instance? or no because there isnt an assignment, I think its fine either way right now because we only create one SideBar instance, but what im wondering if using this in the def __init__() would be more along the lines of what we want:
```
self._buffer_control: BufferControl = BufferControl(...)
```
that way we get the typing, but the variable is only created on the instance of the class

Reply:
I really have no idea what the difference is between the two, you'll have to explain it to me

Reply:
I've woken up now, and am embarassed that I didn't understand that so easily, are you telling me that variables declared at the class level are essentially static variables? That is definitely not what I intended

Reply:
I looked it up, lol, this is almost definitely the source of a couple bugs